ydb/library/yql/sql/settings/serializer/serializer.cpp

```diff
@@ -0,0 +1,57 @@
+#include "serializer.h"
+
+#include <ydb/library/yql/sql/settings/serializer/proto/translation_settings.pb.h>
+
+namespace NSQLTranslation {
+
+void TTranslationSettingsSerializer::Serialize(
+    const TTranslationSettings& settings, NYql::NProto::TTranslationSettings& serializedSettings
+) const {
+    if (PathPrefixSetter) {
+        PathPrefixSetter(*serializedSettings.MutablePathPrefix());
+    } else {
+        serializedSettings.SetPathPrefix(settings.PathPrefix);
+    }
+    serializedSettings.SetSyntaxVersion(settings.SyntaxVersion);
+    serializedSettings.SetAnsiLexer(settings.AnsiLexer);
+    serializedSettings.SetAntlr4Parser(settings.Antlr4Parser);
+    serializedSettings.SetPgParser(settings.PgParser);
+
+    auto* pragmas = serializedSettings.MutablePragmas();
+    pragmas->Add(settings.Flags.begin(), settings.Flags.end());
+}
+
+void TTranslationSettingsSerializer::Serialize(const TTranslationSettings& settings, TString& serializedSettings) const {
+    NYql::NProto::TTranslationSettings protoSettings;
+    Serialize(settings, protoSettings);
+    serializedSettings = protoSettings.SerializeAsString();
+}
+
+void TTranslationSettingsSerializer::Deserialize(
+    const NYql::NProto::TTranslationSettings& serializedSettings, TTranslationSettings& settings
+) const {
+    #define DeserializeSetting(settingName) \
+        if (serializedSettings.Has##settingName()) { \
+            settings.settingName = serializedSettings.Get##settingName(); \
+        }
+
+        DeserializeSetting(PathPrefix);
+        DeserializeSetting(SyntaxVersion);
+        DeserializeSetting(AnsiLexer);
+        DeserializeSetting(Antlr4Parser);
+        DeserializeSetting(PgParser);
+
+    #undef DeserializeSetting
+
+    settings.Flags.insert(serializedSettings.GetPragmas().begin(), serializedSettings.GetPragmas().end());
```

Comment:
#### What if the `Flags` already contain [`AnsiCurrentRow`](https://ydb.tech/docs/en/yql/reference/syntax/pragma#ansicurrentrow) and `serializedSettings` are adding `DisableAnsiCurrentRow`?

See the relevant code [here](https://github.com/ydb-platform/ydb/blob/f38e45016276218330480a40a5575b3d5523522e/ydb/library/yql/sql/v1/context.cpp#L112). I believe, the last pragma will win, i.e. `DisableAnsiCurrentRow` will be in effect.

Takeaway: serialized settings will take precedence over those fields of the TTranslationSettings struct that they are updating.